packages/checkout/filter-registry/index.ts

```diff
@@ -33,6 +33,7 @@ let checkoutFilters: Record<
 	Record< string, CheckoutFilterFunction >
 > = {};
 
+const cachedValues: Record< string, T > = {};
```

Comment:
Why did we remove the use of `ref` here?

Reply:
This thread expands on the problem that we fixed here
https://a8c.slack.com/archives/C8X6Q7XQU/p1663243165986399

In short, mixing a ref for some of the cache and a global variable for other parts of the cache caused a mismatch of cache availability when using the filter in multiple components.

What happened is that each component now had an instance of `cachedValues` yet they operated on the same `cachedFilterRuns` instance, this meant that valid runs are skipped because `cachedFilterRuns` returns false and `cachedValues` is never updated.

This mainly happens when you have 2 instances of `__experimentalApplyCheckoutFilter`, only one of them will get updated, the second one won't.